### PR TITLE
Add doctest for pairs

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/abstractdict.jl [hunk 1]
```diff
@@ -136,6 +136,29 @@ values(a::AbstractDict) = ValueIterator(a)
...
+
+# Examples
+```jldoctest
+julia> a = Dict(zip(["a", "b", "c"], [1, 2, 3]))
+Dict{String, Int64} with 3 entries:
+ "c" => 3
+ "b" => 2
+ "a" => 1
+
+julia> pairs(a)
```
Comment (KristofferC) [inline base/abstractdict.jl:148]: I wonder how useful this specific example is when it literally just returns the input unmodified. How often would you call `pairs` on a `Dict` in practice?

```jl
ulia> (;a=1, b=2) |> pairs |> collect
2-element Vector{Pair{Symbol, Int64}}:
 :a => 1
 :b => 2

julia> (;a=1, b=2) |> collect
2-element Vector{Int64}:
 1
 2
```

might be something that is useful to show?

Comment (kshyatt) [reply base/abstractdict.jl:148]: Sure, why not?

Comment (kshyatt) [reply base/abstractdict.jl:148]: I do think people might naively expect `pairs(d)` to return a `Vector{Pair}` even if `d` is a `Dict`, which is why I included it.

Comment (mcabbott) [reply base/abstractdict.jl:148]: Normally you would iterate this thing, right? So perhaps one of the examples might show that, like
```
julia> foreach(println, pairs(["a", "b", "c"]))
1 => "a"
2 => "b"
3 => "c"
```

**Comment by kshyatt** [comment]
Made changes to address feedback -- good to go?

**Comment by strickek** [comment]
should be `3-element Vector{...}` instead of  `2-element Vector{...}` (2 times)

**Pull request merged**